### ISSUE TITLE
[feature request] new logger contribution

### ISSUE THREAD

**Issue by kamil-kaczmarek** [Stable-Baselines-Team/stable-baselines3-contrib#6]
Hi,

Thanks for the library and all the effort you put into it!

Together with my friend we wanted to contribute new logger to the Stable Baselines, namely neptune-logger.

It will let stable baselines and neptune users automatically log their runs to Neptune (it is [experiment tracking tool](https://docs.neptune.ai/index.html)).

Can I open draft PR, so that we can continue working on it from there?

Please let me know what do you think this idea?

**Comment by araffin** [comment]
Hello,

Thanks for your proposal.
So, if we accept such feature, it must be optional anyway (it should not be required to install neptune in order to use sb3 contrib).
Also, this issue https://github.com/DLR-RM/stable-baselines3/issues/109 needs to be addressed first anyway.

I'm wondering, as it seems that it requires an API key to work, how would test such feature?

**Comment by kamil-kaczmarek** [comment]
Hi @araffin,

> So, if we accept such feature, it must be optional anyway (it should not be required to install neptune in order to use sb3 contrib).

Yes, I agree. We also have Pytorch Lightning integration implemented in their lib. In there, we simply capture exception on import - [see in the code](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/loggers/neptune.py#L22).

> Also, this issue DLR-RM/stable-baselines3#109 needs to be addressed first anyway.

Let me take a closer look at it :slightly_smiling_face: 

> I'm wondering, as it seems that it requires an API key to work, how would test such feature?

That's why we have public API key, that we use in such cases. In this way we will be able to test.

**Comment by araffin** [comment]
>That's why we have public API key, that we use in such cases. In this way we will be able to test.

good to know ;)

**Comment by kamil-kaczmarek** [comment]
Hey @araffin ,

Can you tell a bit about the current status of the https://github.com/DLR-RM/stable-baselines3/issues/109 ? Is these anybody working on it?

Also, I'm curious: I can see that `stable-baselines3-contrib` is under `Stable-Baselines-Team`, while `stable-baselines3` is under the `DLR-RM` organization. Would you mind to explain relationship between those? I'm looking for proper place for proposed integration.

Also, do you think that we should prepare separate integration for `stable-baselines` and `stable-baselines3` ?

**Comment by araffin** [comment]
>Can you tell a bit about the current status of the DLR-RM/stable-baselines3#109 ? Is these anybody working on it?

No one is currrently working on it (hence the "help wanted" tag ;)).

>Also, I'm curious: I can see that stable-baselines3-contrib is under Stable-Baselines-Team, while stable-baselines3 is under the DLR-RM organization. Would you mind to explain relationship between those? I'm looking for proper place for proposed integration.

So, SB3 is under DLR-RM because I mostly built it during my working hours at the German Aerospace Center (DLR).
Stable-Baselines-Team was created to centralize everything under a common domain (and not under a user like `hill-a`) and give all maintainers the same rights.
It concerns things where maintainers use their free time to develop it.
SB3 is the next version of Stable-Baselines (SB2), so you should only focus on SB3.

To sum it up: for the integration, only do it for SB3 on the contrib repo.

**Comment by kamil-kaczmarek** [comment]
@araffin ,

Thanks for the explanation - I'll focus only on SB3-contrib.

Later, this week / early next week I'll take a closer look at https://github.com/DLR-RM/stable-baselines3/issues/109

Thanks

**Comment by kamil-kaczmarek** [comment]
Likely early next week :)

**Comment by araffin** [comment]
Closing this as we would now prefer an external integration as done by W&B and Huggingface, see https://stable-baselines3.readthedocs.io/en/master/guide/integrations.html

**Comment by kamil-kaczmarek** [comment]
hey @araffin 

Got it, thanks for the clarification
